packages/core/contracts/zodiac/OptimisticGovernorCreator.sol

```diff
@@ -0,0 +1,28 @@
+// SPDX-License-Identifier: AGPL-3.0-only
```

Comment:
Our [Lockable](https://github.com/UMAprotocol/protocol/blob/master/packages/core/contracts/common/implementation/Lockable.sol) contract is not exactly the standard OZ reentrancy guard contract. In particular, we have some methods that we have used to change the state of the re-entrancy variable in derived contracts, unlike the OZ one.

My understanding is that we could fix this issue with a one-line change in the OG.

In the initializer/setup method (which isn't re-entrancy guarded I don't think), we could call [this method](https://github.com/UMAprotocol/protocol/blob/855e3c740938f8c6a5675387225d3e059b6c1d70/packages/core/contracts/common/implementation/Lockable.sol#L70-L72) to set the initializer to true:
```
function setUp(bytes memory initializeParams) public override initializer {
  ...
  _startReentrantGuardDisabled();
  ...
}
```

Or, alternatively, you could do:
```
function setUp(bytes memory initializeParams) public override initializer {
  ...
  _postEntranceReset();
  ...
}
```

Both methods do the same thing, we just have different ones for different reasons.

From what I remember, there were advantages to using the proxy pattern in zodiac rather than the factory pattern. If that's the case, this one-line change may be preferable to a factory.

Thoughts @Reinis-FRP @abg4 @smb2796?

Reply:
Hey Matt, thanks for the comment above. I am in favor of doing a small update to the OG contract for it to be able to be compatible with the Zodiac proxy but here are just some of my thoughts on the tradeoffs:

**Zodiac Proxy:**
- I have done testing with and without the proxy and the gas savings are pretty significant when using the Zodiac proxy.
- Being able to use the standard Zodiac deploy/setup functions. We have already done the work to make both work for the OG Zodiac module but I think using the same process as all other Zodiac modules has some benefits.

**Factory Contract:**
- I mentioned in the motivation how the factory pattern makes it easier for tracking deployments / TVL but after looking at the Zodiac proxy contract, we can likely easily achieve this using the proxy contract.
- The networks Zodiac supports is limited so there might be an argument that the factory contract is easier to interact with vs alternative deployment methods in an event where UMA contracts + Snapshot supports a network but Zodiac doesn't?

Reply:
On the points under Factory Contract:
- How would we track the module deployment without a factory contract? Should we make the zodiac contract call out to a separate contract on setup to emit an event?
- Specifically what networks would we be losing out on?

Reply:
For your first question, I was thinking we could just look at the `ModuleProxyCreation` event (https://github.com/gnosis/zodiac/blob/master/contracts/factory/ModuleProxyFactory.sol#L5) emitted by the Zodiac proxy contract and isolate events that use the OG mastercopy address. It would probably make it easier though if we had the Zodiac contract call out to a separate contract on setup to emit an event as you suggested.

If you want to look at an example, [here](https://rinkeby.etherscan.io/address/0x00000000062c52e29e8029Dc2413172f6D619d85#code) is the Zodiac proxy contract on Rinkeby and [here](https://rinkeby.etherscan.io/tx/0xedee3541b78bbd8f79c8c7b5aed4c166e5f16032daf85ad221c6a42832cfe1eb#eventlog) is the logs of an OG contract deployed using the proxy https://rinkeby.etherscan.io/tx/0xedee3541b78bbd8f79c8c7b5aed4c166e5f16032daf85ad221c6a42832cfe1eb#eventlog.

For the second point, the Zodiac supported networks are Mainnet, Rinkeby, Binance smart chain, Gnosis chain, Polygon, and Mumbai. I think I need to look more into what using Snapshot with the OG on a network not supported by Zodiac would require. It might be better to just ignore this point as there are various ways we could make deployments on these other networks easier if a factory pattern isn't used.

Reply:
@abg4 I agree with sentiment of preferring OG update over factory.

> Being able to use the standard Zodiac deploy/setup functions. We have already done the work to make both work for the OG Zodiac module but I think using the same process as all other Zodiac modules has some benefits

I think this is a great point and shouldn't be understated. There are likely a lot of long term benefits from conforming to zodiac standards.

 > For the second point, the Zodiac supported networks are Mainnet, Rinkeby, Binance smart chain, Gnosis chain, Polygon, and Mumbai. I think I need to look more into what using Snapshot with the OG on a network not supported by Zodiac would require.

As we are building an e2e product that relies entirely on UX of zodiac & snapshot, I think we shouldn't really be trying to support chains that are not supported by zodiac anyways. Like you said, seems fine to figure this out for other networks as needed (even though it probably shouldn't be).

Reply:
TLDR: I think we should move forward with the one-line change in the OG. @abg4 up to you whether we do that in this PR or if you want to close this and open a separate PR.

> For your first question, I was thinking we could just look at the ModuleProxyCreation event (https://github.com/gnosis/zodiac/blob/master/contracts/factory/ModuleProxyFactory.sol#L5) emitted by the Zodiac proxy contract and isolate events that use the OG mastercopy address. It would probably make it easier though if we had the Zodiac contract call out to a separate contract on setup to emit an event as you suggested.

If the same event is always emitted by the same contract, it should be super easy to filter.

> As we are building an e2e product that relies entirely on UX of zodiac & snapshot, I think we shouldn't really be trying to support chains that are not supported by zodiac anyways. Like you said, seems fine to figure this out for other networks as needed (even though it probably shouldn't be).

Agree, SGTM.